### DISCUSSION TITLE
Duda con ejercicio N°36 TP While

### DISCUSSION THREAD

**Discussion by cdv13** [santiemanuel/milProgramadoresQA#10] [Preguntas y Respuestas] [answered]
Hola, tengo una duda con la primera parte del ejercicio N°36 del TP While 

_"Realizar un programa que permita ingresar repetidamente 3 nombres de personas determinar si los nombres no son iguales"_

¿Como se comparan los valores de la variable cuando son ingresadas mediante un ciclo?. O lo estoy interpretando mal?

**Comment by santiemanuel** [comment, marked as answer]
Hola, lo que dice es que en cada iteración se pedirán 3 nombres (3 variables distintas).
- Si no son iguales inicia un nuevo ciclo pidiendo 3 notas y calculando el promedio de estas 3.
- Si son iguales, pido un número e imprimo la tabla de multiplicar de este número

Bueno, y finalmente habla sobre contar ocurrencias de los disferentes casos e indica la condición de corte del ciclo principal (calcular el promedio 5 veces).

**Reply by cdv13** [reply]
Si comprendí el proceso, pero no sabia como comparar los valores (nombres) luego de ingresarlos por el ciclo.
después de buscar en la red, la única solución que vi para poder comparar esos valores, es usando una lista.....Hay otra forma?

comparto como me quedo el código:

```py
i=0
n=0
suma=0
promedio=0
mu=0
pr=0
k=1
i=0
nombres2=[]

while i<3:
  nombre=input("Ingrese un nombre: ")
  i+=1
  nombres2.append(nombre)

print(nombres2)

ele=nombres2[0]
resultado=True

for item in nombres2:
  if ele != item:
    resultado=False
    break
if (resultado==True):
  print("Iguales")
  n=int(input("Ingrese un numero: "))
  for k in range(1,11):
    multiplicacion=n*k
    print(f"{n} x {k} = {multiplicacion}")
  mu+=1
  
else:
  print("Desiguales")
  while n<3:
      nota=int(input("Ingrese las notas: "))
      n+=1
      suma+=nota
      promedio=suma/n
  pr+=1
  print(f"El promedio es de: {promedio}")
 
print("Se calculo ",pr,"veces el promedio, y se motro ",mu,"veces la tabla")
```

**Reply by santiemanuel** [reply]
Si vas a usar listas podés usar los conjuntos (set), que básicamente no repiten sus elementos, entonces, si agregas los tres nombres y son iguales len(conjunto) va a ser igual a 1, si no, puede ser 2 o 3.

Entonces preguntas si len(conjunto) == 1, hago el proceso para los tres nombres iguales, else, para desiguales.

Para inicializar un conjunto vacío haces `nombres = set()`, para agregar un nombre hacés `nombres.add(cadena)`
